src/common/meta/src/kv_backend/etcd.rs

```diff
@@ -33,55 +33,39 @@ use crate::rpc::store::{
 };
 use crate::rpc::KeyValue;
 
-pub struct KvPair<'a>(&'a etcd_client::KeyValue);
-
-impl<'a> KvPair<'a> {
-    /// Creates a `KvPair` from etcd KeyValue
-    #[inline]
-    pub fn new(kv: &'a etcd_client::KeyValue) -> Self {
-        Self(kv)
-    }
-
-    #[inline]
-    pub fn from_etcd_kv(kv: &etcd_client::KeyValue) -> KeyValue {
-        KeyValue::from(KvPair::new(kv))
-    }
-}
-
-impl<'a> From<KvPair<'a>> for KeyValue {
-    fn from(kv: KvPair<'a>) -> Self {
-        Self {
-            key: kv.0.key().to_vec(),
-            value: kv.0.value().to_vec(),
-        }
-    }
-}
-
 // Maximum number of operations permitted in a transaction.
 // The etcd default configuration's `--max-txn-ops` is 128.
 //
 // For more detail, see: https://etcd.io/docs/v3.5/op-guide/configuration/
 const MAX_TXN_SIZE: usize = 128;
 
 pub struct EtcdStore {
+    root: Vec<u8>,
     client: Client,
```

Comment:
Another consideration is that maybe it's an alternative to wrap `Client` since we have also `EtcdLock` and `EtcdElection` that use `Client`.

Reply:
No. It's hard to intercept all the layers. Prepend/Strip keys in place is more desired.

Reply:
Since it's internal implementation, we can also postpone the decision until we really meet troubles. I have the idea to convert to that solution on demand.